src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,118 @@
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * Storage of data from the user specified relative file path.
+ * CS2103T iP
+ * AY22/23 Semester 1
+ *
+ * @author Perry Wong
+ */
+public class Storage {
+
+    private String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    private boolean isMarked(String s) {
+        return s.equals("X");
+    }
+
+    private void createFile(File f) throws IOException {
+        f.getParentFile().mkdirs();
+        f.createNewFile();
+    }
+
+
+    /**
+     * Loads tasks from the data file if the file path exists.
+     *
+     * @return List of tasks that was saved previously.
+     * @throws IOException If there is an issue with accessing the file.
+     * @throws DukeException If Duke is unable to process the contents of the file as they are invalid data.
+     */
+    public ArrayList<Task> load() throws IOException, DukeException {
+        File file = new File(filePath);
+        if (!file.exists()) {
+            this.createFile(file);
+            throw new DukeException("Loading error.");
+        } else {
+            System.out.println("Saved tasks retrieved!\n");
+            ArrayList<Task> tasks = new ArrayList<>(100);
+            Scanner sc = new Scanner(file);
+
+            while (sc.hasNextLine()) {
+                String text = sc.nextLine();
+                Scanner temp = new Scanner(text);
+                temp.useDelimiter("\\[|\\]\\s*|by:\\s*|at:\\s*|\\s*\\(|\\s*\\)");
+
+                String type = temp.next();
+                String ignore = temp.next();
+                String marked = temp.next();
+                String description = temp.next();
+
+                if (type.equals("D")) {
+                    String ignore1 = temp.next();
+                    String date = temp.next();
+                    DateTimeFormatter alt = DateTimeFormatter.ofPattern("MMM d yyyy");
+                    DateTimeFormatter input = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+                    String dateAlt = input.format(alt.parse(date));
+                    Deadline d = new Deadline(description, dateAlt);
+                    if (isMarked(marked)) {
+                        d.mark();
+                    }
+                    tasks.add(d);
+                } else if (type.equals("E")) {
+                    String ignore1 = temp.next();
+                    String date = temp.next();
+                    DateTimeFormatter alt = DateTimeFormatter.ofPattern("MMM d yyyy");
+                    DateTimeFormatter input = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+                    String dateAlt = input.format(alt.parse(date));
+                    Deadline d = new Deadline(description, dateAlt);
+                    Event e = new Event(description, dateAlt);
```

Comment:
Could the naming be more meaningful? I think it would be better to give a more descriptive naming to avoid confusion later on.

Reply:
Assuming you are referring to `dateAlt`, I'll try to think of a more appropriate name. Thanks!